testdata/templates/args.yaml

```diff
@@ -0,0 +1,8 @@
+service: foo
+headers:
+  header1: ${user:foo}
```

Comment:
Sounds reasonable. Although `mapdecode` expects to decode into a known struct shape, not dynamic Thrift shapes.